### PR TITLE
ZS2022-33 Added tests for Audit

### PR DESCRIPTION
**Pull request by DariusChis**
https://zenitech.atlassian.net/browse/ZS2022-33

### CODE REVIEW

File: src/main/java/pageObjects/Audit.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        waitAuditTableOpen();
+
+    }
+
+    public void successfullyEdicDeviceAudit() {
+        devicePage.successfullyEditADeviceSerialNo();
+        waitAuditTableOpen();
+    }
+
+    private void waitAuditTableOpen(){
```
Comment (CaprarClaudiu) [inline src/main/java/pageObjects/Audit.java]: this looks like an helper method, you can put it in DriverFactory for now. also i think we could use a better name, waitForAuditDisplay or something like that

File: src/main/java/pageObjects/Audit.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        waitAuditTableOpen();
+    }
+
+    public void successfullyUnAssignDeviceAudit() {
+        devicePage.succesfullyUnAssignDevice();
+        waitAuditTableOpen();
+
+    }
+
+    public void successfullyEdicDeviceAudit() {
```
Comment (CaprarClaudiu) [inline src/main/java/pageObjects/Audit.java]: :)

File: src/test/java/pageObjects/AuditTests.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        Assert.assertTrue(tableText.contains("UNASSIGN"));
+    }
+
+    @Test
+    public void successfullyAuditForEditDevice() {
+        audit.successfullyEdicDeviceAudit();
+        String tableText = driver.findElement(By.className("MuiCollapse-wrapper")).getText();
+        Assert.assertTrue(tableText.contains("EDIT"));
+    }
+}
```
Comment (CaprarClaudiu) [inline src/test/java/pageObjects/AuditTests.java]: leave an empty row at the end of the file

File: src/test/java/pageObjects/AuditTests.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        String tableText = driver.findElement(By.className("MuiCollapse-wrapper")).getText();
+        Assert.assertTrue(tableText.contains("CREATE"));
+    }
+
+    @Test
+    public void successfullyAuditForAssignDevice() {
+        audit.successfullyAssignDeviceAudit();
+        String tableText = driver.findElement(By.className("MuiCollapse-wrapper")).getText();
+        Assert.assertTrue(tableText.contains("ASSIGN"));
+    }
```
Comment (CaprarClaudiu) [inline src/test/java/pageObjects/AuditTests.java:25]: do we know to who the device is assigned to? if we do, we can also have a check to see if that persons name is displayed in the audit

**Review by CaprarClaudiu — CHANGES_REQUESTED** [review]

**Review by CaprarClaudiu — APPROVED** [review]

File: src/main/java/pageObjects/Audit.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package pageObjects;
+
+import driver.DriverFactory;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
```
Comment (CaprarClaudiu) [inline src/main/java/pageObjects/Audit.java]: :)

**Review by CaprarClaudiu — CHANGES_REQUESTED** [review]

**Review by ScorusMihai — APPROVED** [review]

**Review by CaprarClaudiu — APPROVED** [review]

**Pull request merged**